test/lib/release_notes_fetchers/url_test.rb

```diff
@@ -0,0 +1,275 @@
+          fetcher.release_notes(prefs)
+        end
+      end
+    end
+
+    context "when release notes are not found" do
+      let(:curl_retcode) { 1 }
+
+      it "blacklists the URL" do
+        expect(described_class).to receive(:add_to_blacklist).with(relnotes_url)
+        fetcher.release_notes(prefs)
+      end
+    end
+
+    context "when a connection problem happens" do
+      let(:curl_retcode) { 5 }
+
+      it "disables downloading release notes via relnotes_url" do
+        expect(described_class).to receive(:disable!).and_call_original
+        fetcher.release_notes(prefs)
+      end
+    end
+
+    context "when release notes URL is not valid" do
+      let(:relnotes_url) { "http" }
+
+      it "returns nil" do
+        expect(fetcher.release_notes(prefs)).to be_nil
+      end
+    end
+
+    context "when release notes URL is nil" do
+      let(:relnotes_url) { nil }
+
+      it "returns nil" do
+        expect(fetcher.release_notes(prefs)).to be_nil
+      end
+    end
+
+    context "when release notes URL is empty" do
+      let(:relnotes_url) { "" }
+
+      it "returns nil" do
+        expect(fetcher.release_notes(prefs)).to be_nil
+      end
+    end
+
+    context "when relnotes_url is blacklisted" do
+      before do
+        described_class.add_to_blacklist(relnotes_url)
+      end
+
+      it "returns nil" do
+        expect(fetcher.release_notes(prefs)).to be_nil
+      end
+
+      it "does not tries to download anything" do
+        expect(Yast::SCR).to_not receive(:Execute)
+          .with(Yast::Path.new(".target.bash"), /curl/)
+        fetcher.release_notes(prefs)
+      end
+    end
+
+    context "when release notes downloading is disabled" do
+      before do
+        described_class.disable!
+      end
+
+      it "returns nil" do
+        expect(fetcher.release_notes(prefs)).to be_nil
+      end
+
+      it "does not tries to download anything" do
+        expect(Yast::SCR).to_not receive(:Execute)
+          .with(Yast::Path.new(".target.bash"), /curl/)
+        fetcher.release_notes(prefs)
+      end
+    end
+
+    context "when a proxy is needed " do
+      let(:proxy_enabled) { true }
+
+      before do
+        allow(Yast::Proxy).to receive(:http).twice.and_return("http://proxy.example.com")
+        allow(Yast::Proxy).to receive(:user).and_return(proxy_user)
+        allow(Yast::Proxy).to receive(:pass).and_return(proxy_pass)
+        test = {
+          "HTTP" => {
+            "tested" => true,
+            "exit"   => 0
+          }
+        }
+        allow(Yast::Proxy).to receive(:RunTestProxy).and_return(test)
+      end
+
+      context "and no user or password are specified" do
+        let(:proxy_user) { "" }
+        let(:proxy_pass) { "" }
+
+        it "uses an unauthenticated proxy" do
+          expect(Yast::SCR).to receive(:Execute) do |_path, cmd|
+            expect(cmd).to include("--proxy http://proxy.example.com")
```

Comment:
The `--proxy` option is used also with an authenticated proxy (see the test below). The correct test would be the `--proxy` option is present and the `--proxy-user` option is **not** present.